mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulProfileScreenTest.kt

```diff
@@ -97,4 +100,43 @@ class StatefulProfileScreenTest {
     rule.onNodeWithText(strings.profileChallenge.uppercase()).performClick()
     rule.onNode(hasText("user2") and hasClickAction()).assertIsSelected()
   }
+
+  @Test
+  fun given_userIsLoggedIn_when_clickedOnUnfollowFriend_then_theButtonShouldChangeToFollow() =
+      runTest {
+    val assets = emptyAssets()
+
+    val auth = buildAuth { user("email@example.org", "password", "1") }
+    val store = buildStore {
+      collection("users") { document("userId2", ProfileDocument(uid = "userId2", name = "user2")) }
+    }
+
+    val authFacade = AuthenticationFacade(auth, store)
+    val chessFacade = ChessFacade(auth, store, assets)
+    val socialFacade = SocialFacade(auth, store)
+    val speechFacade = SpeechFacade(FailingSpeechRecognizerFactory)
+    val tournamentFacade = TournamentFacade(auth, store)
+
+    authFacade.signUpWithEmail("user1@email", "user1", "password")
+    val authUser1 = authFacade.currentUser.filterIsInstance<AuthenticatedUser>().first()
+    val user2 =
+        socialFacade.profile(uid = "userId2", user = authUser1).filterIsInstance<Profile>().first()
+    authUser1.follow(user2)
+
+    val strings =
+        rule.setContentWithLocalizedStrings {
+          PawniesTheme {
+            ProvideFacades(authFacade, socialFacade, chessFacade, speechFacade, tournamentFacade) {
+              Navigation()
+            }
+          }
+        }
+
+    rule.onNodeWithText(strings.sectionSocial).performClick()
+    rule.onNodeWithText("user2").performClick()
+    rule.onNodeWithText(strings.profileUnfollow).performClick()
+    rule.onNodeWithText(strings.profileFollow).assertExists()
+    rule.onNodeWithText(strings.profileFollow).performClick()
+    rule.onNodeWithText(strings.profileUnfollow).assertExists()
```

Comment:
For these kind of "simple" tests, you should use the new `rule.setContentWithTestEnvironment`. You could pass it a custom store (instead of its default one) where you specify that `user2`  is followed by `authUser1` (or rather the default value/id defined in `test/state/Dsk.kt`) directly in the `ProfileDocument`.

Reply:
thx, i've changed it

Reply:
thx, i've changed it